dd-java-agent/instrumentation/kafka-clients-0.11/src/main/java/datadog/trace/instrumentation/kafka_clients/KafkaDecorator.java

```diff
@@ -59,6 +61,10 @@ public void onConsume(final AgentSpan span, final ConsumerRecord record) {
       span.setTag(DDTags.RESOURCE_NAME, "Consume Topic " + topic);
       span.setTag("partition", record.partition());
       span.setTag("offset", record.offset());
+
+      final long produceTime = record.timestamp();
+      final long consumeTime = TimeUnit.NANOSECONDS.toMillis(span.getLocalRootSpan().getStartTime());
```

Comment:
```suggestion
      final long consumeTime = TimeUnit.NANOSECONDS.toMillis(span.getStartTime());
```

Semantically, the span passed into the method is the one representing the consuming of the message.  Besides, it will always be the root span anyway.